File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    alpha = {'zero':'0', 'one':'1', 'two':'2', 'three':'3', 'four':'4',
             'five':'5', 'six':'6', 'seven':'7', 'eight':'8', 'nine':'9'}
    answer = []
    temp = []
    
    for v in s:
        if v.isalpha():
            temp.append(v)
        else:
            answer.append(v)
            
        key = ''.join(temp)
        if key in alpha:
            answer.append(alpha[key])
            temp.clear()
        
    return int(''.join(answer))

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        StringBuilder sb = new StringBuilder(s);
        String[] words = new String[] {
            "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"
        };
        
        for (int i=0; i<words.length; i++) {
            int idx = sb.indexOf(words[i]);
            
            while (idx != -1) {
                sb.replace(idx, idx + words[i].length(), Integer.toString(i));
                idx = sb.indexOf(words[i], idx+1);
            }
        }
        
        return Integer.parseInt(sb.toString());
    }
}